crates/database/src/settlement_observations.rs

```diff
@@ -0,0 +1,146 @@
+use {crate::events::EventIndex, bigdecimal::BigDecimal, sqlx::PgConnection};
+
+pub async fn get_settlement_event_without_observation(
+    ex: &mut PgConnection,
+    max_block_number: i64,
+) -> Result<Option<EventIndex>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT block_number, log_index
+FROM settlement_observations
+WHERE gas_used IS NULL AND block_number <= $1
+LIMIT 1
+    "#;
+    sqlx::query_as(QUERY)
+        .bind(max_block_number)
+        .fetch_optional(ex)
+        .await
+}
+
+#[derive(Debug, Default, PartialEq, sqlx::FromRow)]
+pub struct Observation {
+    pub gas_used: Option<BigDecimal>,
+    pub effective_gas_price: Option<BigDecimal>,
+    pub surplus: Option<BigDecimal>,
+    pub fee: Option<BigDecimal>,
+    pub block_number: i64,
+    pub log_index: i64,
+}
+
+pub async fn update(
+    ex: &mut PgConnection,
+    block_number: i64,
+    log_index: i64,
+    gas_used: BigDecimal,
+    effective_gas_price: BigDecimal,
+    surplus: BigDecimal,
+    fee: BigDecimal,
```

Comment:
Might be nice to group this into an `SettlementObservation` struct.

Reply:
Fixed.